src/transformers/pipelines/__init__.py

```diff
@@ -546,9 +593,24 @@ def pipeline(
         task = get_task(model, use_auth_token)
 
     # Retrieve the task
-    targeted_task, task_options = check_task(task)
-    if pipeline_class is None:
-        pipeline_class = targeted_task["impl"]
+    if task in custom_tasks:
+        targeted_task, task_options = clean_custom_task(custom_tasks[task])
+        if pipeline_class is None:
+            if not trust_remote_code:
+                raise ValueError(
+                    f"Loading this pipeline requires you to execute the code in the pipeline file in that"
+                    f" repo on your local machine. Make sure you have read the code there to avoid malicious use, then"
+                    f" set the option `trust_remote_code=True` to remove this error."
```

Comment:
```suggestion
                    "Loading this pipeline requires you to execute the code in the pipeline file in that"
                    " repo on your local machine. Make sure you have read the code there to avoid malicious use, then"
                    " set the option `trust_remote_code=True` to remove this error."
```